### ISSUE TITLE
Multiple Apache Virtual Hosts

### ISSUE THREAD

**Issue by althe3rd** [puphpet/puphpet#1311]
When creating a configuration in puphpet that involves two or more virtual hosts I noticed that upon running vagrant up and examining the .conf files via ssh that all of the .conf files except for the first virtual host are missing all of their configuration details (such as AllowOveride All, etc.).

So if you were to make a configuration (ubuntu 14) with awesome.dev and fantastic.dev both with options of AllowOverride All set, only awesome.dev will have that value applied to its conf file.

For my development I have manually fixed my files but I could see this being a long standing issue for anyone creating configurations in puphpet that are using default values and aren't working.

I have tested this on 6 different workstations with the same results.

Update: I realize that there may be some underlying issue with how the configuration is being ran in puppet, so please don't assume I place the blame entirely on puphpet.

**Comment by althe3rd** [comment]
In case it helps I have included the config.yaml file generated via the puphpet website that I was using.  Everything completed correctly according to the console, but the second virtual host conf file never gets the correct flags set.


---
vagrantfile-local:
    vm:
        box: puphpet/ubuntu1404-x32
        box_url: puphpet/ubuntu1404-x32
        hostname: ''
        memory: '512'
        cpus: '1'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.56.102
            forwarded_port:
                8cfVqGQUftdG:
                    host: '5385'
                    guest: '22'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: on
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            L5OnjAd1vu6a:
                owner: www-data
                group: www-data
                source: ./
                target: /var/www
                sync_type: default
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                    auto: 'false'
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
server:
    install: '1'
    packages: {  }
users_groups:
    install: '1'
    groups: {  }
    users: {  }
cron:
    install: '1'
    jobs: {  }
firewall:
    install: '1'
    rules: null
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - rewrite
    vhosts:
        gdiik5vaxtur:
            servername: calsmain.localhost
            serveraliases:
                - www.calsmain.localhost
                - '*.calsmain.localhost'
            docroot: /var/www/calsmain
            port: '80'
            setenv:
                - 'APP_ENV dev'
            directories:
                f8yb8sm8cr3p:
                    provider: directory
                    path: /var/www/calsmain
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - all
                        - granted
                    custom_fragment: ''
            engine: php
            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
        o1ttczqlflc1:
            servername: hostcals.localhost
            serveraliases:
                - www.hostcals.localhost
            docroot: /var/www/hostcals
            port: '80'
            directories:
                eojscrnmlqwv:
                    provider: directory
                    path: ''
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - all
                        - granted
                    custom_fragment: ''
            engine: php
            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
    mod_pagespeed: 0
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    upstreams: {  }
    vhosts:
        rtwjldvmml1i:
            proxy: ''
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            location: \.php$
            index_files:
                - index.html
                - index.htm
                - index.php
            envvars:
                - 'APP_ENV dev'
            engine: php
            client_max_body_size: 1m
            ssl_cert: ''
            ssl_key: ''
php:
    install: '1'
    version: '55'
    composer: '1'
    composer_home: ''
    modules:
        php:
            - cli
            - intl
            - mcrypt
        pear: {  }
        pecl:
            - pecl_http
    ini:
        display_errors: On
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
    timezone: America/Chicago
    mod_php: '1'
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings:
        host: 127.0.0.1
        port: '9000'
    ini:
        display_errors: On
        error_reporting: '-1'
    timezone: null
xdebug:
    install: '0'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.17.1
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions:
        tNc0oI7ZWdWV:
            version: ''
nodejs:
    install: '0'
    npm_packages: {  }
python:
    install: '1'
    packages: {  }
    versions:
        2RPF198QZdmL:
            version: ''
mysql:
    install: '1'
    override_options: {  }
    root_password: '123'
    adminer: 0
    databases:
        uU62tGw8GMZO:
            grant:
                - ALL
            name: calsmaindb
            host: localhost
            user: calsmaindbuser
            password: '123'
            sql_file: /var/www/sql/calsmain.sql
        BBGqKbPCF4I3:
            grant:
                - ALL
            name: hostcalsdb
            host: localhost
            user: hostcalsdbsuser
            password: '123'
            sql_file: /var/www/sql/hostcals.sql
postgresql:
    install: '0'
    settings:
        root_password: '123'
        user_group: postgres
        encoding: UTF8
        version: '9.3'
    databases: {  }
    adminer: 0
mariadb:
    install: '0'
    override_options: {  }
    root_password: '123'
    adminer: 0
    databases: {  }
    version: '10.0'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mongodb:
    install: '0'
    settings:
        auth: 1
        port: '27017'
    databases: {  }
redis:
    install: '0'
    settings:
        conf_port: '6379'
mailcatcher:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '1080'
        mailcatcher_path: /usr/local/rvm/wrappers/default
        from_email_method: inline
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '13000'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
    binlogdir: /var/lib/beanstalkd/binlog
rabbitmq:
    install: '0'
    settings:
        port: '5672'
elastic_search:
    install: '0'
    settings:
        version: 1.4.1
        java_install: true
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'

**Comment by vaccinemedia** [comment]
I'm seeing the same behaviour apart from all my standard apache virtual hosts have the options and AllowOverides set apart from the ones which are set to use ssl

**Comment by jtreminio** [comment]
Using the following config:

```
---
vagrantfile-local:
    vm:
        box: puphpet/ubuntu1404-x32
        box_url: puphpet/ubuntu1404-x32
        hostname: ''
        memory: '512'
        cpus: '1'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.56.176
            forwarded_port:
                gNJ8e3wWgNu3:
                    host: '9086'
                    guest: '22'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: on
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            uKqmRnTekX0m:
                owner: www-data
                group: www-data
                source: ./
                target: /var/www
                sync_type: default
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                    auto: 'false'
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
server:
    install: '1'
    packages:
        - vim
users_groups:
    install: '1'
    groups: {  }
    users: {  }
cron:
    install: '1'
    jobs: {  }
firewall:
    install: '1'
    rules: null
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - rewrite
    vhosts:
        bqaiiz7j6rhm:
            servername: awesome.dev
            serveraliases:
                - www.awesome.dev
            docroot: /var/www/awesome
            port: '80'
            setenv:
                - 'APP_ENV dev'
            directories:
                vhtfjb9frww2:
                    provider: directory
                    path: /var/www/awesome
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - all
                        - granted
                    custom_fragment: ''
            engine: php
            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
        foo:
            servername: fantastic.dev
            serveraliases:
                - www.fantastic.dev
            docroot: /var/www/fantastic
            port: '80'
            setenv:
                - 'APP_ENV dev'
            directories:
                vhtfjb9frww2:
                    provider: directory
                    path: /var/www/fantastic
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - all
                        - granted
                    custom_fragment: ''
            engine: php
            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
    mod_pagespeed: 0
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    upstreams: {  }
    vhosts:
        a304efpech7l:
            proxy: ''
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            location: \.php$
            index_files:
                - index.html
                - index.htm
                - index.php
            envvars:
                - 'APP_ENV dev'
            engine: php
            client_max_body_size: 1m
            ssl_cert: ''
            ssl_key: ''
php:
    install: '0'
    version: '56'
    composer: '1'
    composer_home: ''
    modules:
        php:
            - cli
            - intl
            - mcrypt
        pear: {  }
        pecl:
            - pecl_http
    ini:
        display_errors: On
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
    timezone: America/Chicago
    mod_php: 0
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings:
        host: 127.0.0.1
        port: '9000'
    ini:
        display_errors: On
        error_reporting: '-1'
    timezone: null
xdebug:
    install: '0'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.17.1
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions:
        yvB9Dfpi1MiK:
            version: ''
nodejs:
    install: '0'
    npm_packages: {  }
python:
    install: '1'
    packages: {  }
    versions:
        2cXdjHaIg1dS:
            version: ''
mysql:
    install: '0'
    override_options: {  }
    root_password: '123'
    adminer: 0
    databases:
        Omc1rfNgiaUT:
            grant:
                - ALL
            name: dbname
            host: localhost
            user: dbuser
            password: '123'
            sql_file: ''
postgresql:
    install: '0'
    settings:
        root_password: '123'
        user_group: postgres
        encoding: UTF8
        version: '9.3'
    databases: {  }
    adminer: 0
mariadb:
    install: '0'
    override_options: {  }
    root_password: '123'
    adminer: 0
    databases: {  }
    version: '10.0'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mongodb:
    install: '0'
    settings:
        auth: 1
        port: '27017'
    databases: {  }
redis:
    install: '0'
    settings:
        conf_port: '6379'
mailcatcher:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '1080'
        mailcatcher_path: /usr/local/rvm/wrappers/default
        from_email_method: inline
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '13000'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
    binlogdir: /var/lib/beanstalkd/binlog
rabbitmq:
    install: '0'
    settings:
        port: '5672'
elastic_search:
    install: '0'
    settings:
        version: 1.4.1
        java_install: true
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

I see:

```
[07:56 PM]-[vagrant@packer-virtualbox-iso]-[/etc/apache2/sites-enabled]
$ cat 25-bqaiiz7j6rhm.conf
# ************************************
# Vhost template in module puppetlabs-apache
# Managed by Puppet
# ************************************

<VirtualHost *:80>
  ServerName awesome.dev

  ## Vhost docroot
  DocumentRoot "/var/www/awesome"



  ## Directories, there should at least be a declaration for /var/www/awesome

  <Directory "/var/www/awesome">
    Options Indexes FollowSymlinks MultiViews
    AllowOverride All
    Require all granted

  </Directory>

  ## Load additional static includes


  ## Logging
  ErrorLog "/var/log/apache2/bqaiiz7j6rhm_error.log"
  ServerSignature Off
  CustomLog "/var/log/apache2/bqaiiz7j6rhm_access.log" combined




  ## Server aliases
  ServerAlias www.awesome.dev

  ## SetEnv/SetEnvIf for environment variables
  SetEnv APP_ENV dev

  ## Custom fragment





    ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi:///var/www/awesome/$1




</VirtualHost>
```

and 

```
[07:57 PM]-[vagrant@packer-virtualbox-iso]-[/etc/apache2/sites-enabled]
$ cat 25-foo.conf
# ************************************
# Vhost template in module puppetlabs-apache
# Managed by Puppet
# ************************************

<VirtualHost *:80>
  ServerName fantastic.dev

  ## Vhost docroot
  DocumentRoot "/var/www/fantastic"



  ## Directories, there should at least be a declaration for /var/www/fantastic

  <Directory "/var/www/fantastic">
    Options Indexes FollowSymlinks MultiViews
    AllowOverride All
    Require all granted

  </Directory>

  ## Load additional static includes


  ## Logging
  ErrorLog "/var/log/apache2/foo_error.log"
  ServerSignature Off
  CustomLog "/var/log/apache2/foo_access.log" combined




  ## Server aliases
  ServerAlias www.fantastic.dev

  ## SetEnv/SetEnvIf for environment variables
  SetEnv APP_ENV dev

  ## Custom fragment





    ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi:///var/www/fantastic/$1




</VirtualHost>
```

**Comment by althe3rd** [comment]
Thanks @jtreminio !  I'll give that new config a try in just a few moments and let you know what I find.  If things aren't coming out the same on my end for the virtual host files then perhaps that helps me narrow the problem down to something on my version of vagrant or some other component.

But I bet that config you posted will work just fine.

**Comment by althe3rd** [comment]
The new config.yaml file you included @jtreminio worked perfectly.  

I noticed that when I looked over the difference of the two config files that the directory for the second virtual host was empty.  So I regenerated another config via the puphpet website to be sure that value was entered there and sure enough when i got the new config.yaml file there it was once again empty.

Thankfully its not difficult to just update the config.yaml file manually, just figured I would share that experience in case it helps solve a but with the web app generating the config files.
